src_python/habitat_sim/utils/blender_load_replay.py

```diff
@@ -0,0 +1,233 @@
+try:
+    import bpy
+except ImportError:
+    raise ImportError(
+        "Failed to import Blender modules. This script can't run "
+        "standalone. Run `blender --python path/to/blender_load_replay.py ...`. Watch the terminal for "
+        "debug/error output."
+    )
+
+import argparse
+import json
+import os
+from dataclasses import dataclass
+
+
+@dataclass
+class ImportItem:
+    filepath: str
+    do_join_all: bool = True
+    force_color: list = None
+
+
+def import_scene_helper(raw_filepath):
+    # these fixups will be tried one at a time, in order, until a matching, existing file is found
+    model_filepath_fixups = [
+        # use uncompressed versions when available
+        ("/stages/", "/stages_uncompressed/"),
+        ("/urdf/", "/urdf_uncompressed/"),
+        # ("./data/robots/hab_fetch/robots/../meshes/", "/home/eundersander/projects/blender/fetch_meshes_modified/")
+        # for Fetch, avoid blender .dae import because it seems to be broken; fall back to collision STL
+        (".dae", "_collision.STL"),
+        # sometimes the collision STL doesn't have _collision in the name
+        (".dae", ".STL"),
+        # the null fixup (use model filepath as-is)
+        None,
+    ]
+
+    filepath = None
+    for fixup in model_filepath_fixups:
+        if fixup is None:
+            filepath = raw_filepath
+        else:
+            if raw_filepath.find(fixup[0]) != -1:
+                filepath = raw_filepath.replace(fixup[0], fixup[1])
+            else:
+                continue
+        if os.path.exists(filepath):
+            break
+    else:
+        raise RuntimeError("can't find file " + raw_filepath)
+
+    ext = os.path.splitext(filepath)[1].lower()
+    if ext == ".glb" or ext == ".gltf":
+        filename = os.path.basename(filepath)
+        bpy.ops.import_scene.gltf(
+            filepath=filepath, files=[{"name": filename, "name": filename}], loglevel=50
+        )
+    elif ext == ".obj":
+        bpy.ops.import_scene.obj(filepath=filepath)
+    elif ext == ".dae":
+        bpy.ops.wm.collada_import(filepath=filepath)
+    elif ext == ".stl":  # noqa: SIM106
+        bpy.ops.import_mesh.stl(filepath=filepath)
+    else:
+        raise RuntimeError("no importer found for " + filepath)
+
+    return filepath
+
+
+def import_item(item):
+
+    adjusted_filepath = import_scene_helper(item.filepath)
+
+    bpy.ops.object.select_all(action="SELECT")
+    bpy.context.view_layer.objects.active = bpy.context.selected_objects[0]
+
+    childless_empties = [
+        e
+        for e in bpy.context.selected_objects
+        if e.type.startswith("EMPTY") and not e.children
+    ]
+    if len(childless_empties):
+        print("removing {} childless EMPTY nodes".format(len(childless_empties)))
```

Comment:
logger

Reply:
I think you are talking about our Habitat logger, `from habitat.core.logging import logger`.

We don't want this script to depend on other scripts, if possible. I.e. don't import anything from the rest of Habitat. For example, we may want to share this script with a tech artist in the future by Slacking/emailing the file (since tech artists won't want to clone our repo).

I looked for a "standard" way to issue non-exceptional warnings in Python and found the `warnings.warn` module, and it could be used here, but its output is quite verbose and not consistent with other Blender python output, e.g. how the built-in importers print warnings.

I think we should stick with `print`.